### PR TITLE
modules: Kconfig: Provide default definitions for modules with blobs

### LINKED ISSUE #78056: check_compliance.py test fails if environment have not synced a extra vendors
**Describe the bug**
Before #77386, `./script/ci/check_compliance.py` would run without the need of extra HAL modules. After that PR, the script now requires the module sync for vendors with blobs, as such as Espressif, STM32, Infineon and NXP.

This degrades CI performance as it can't be optimized for custom CIs. 

**To Reproduce**
1) Sync with latest main
2) `west update`
3) Delete one of the hal folder vendors describe above: `rm -rf zephyr/modules/hal/xxxx`
4) `./script/ci/check_compliance.py`

**Expected behavior**
Compliance check to work without any issues.

**Impact**

**Logs and console output**
If stm32 hal is removed for example:
```
warning: undefined symbol ZEPHYR_HAL_STM32_MODULE_BLOBS:

- Referenced at drivers/bluetooth/hci/Kconfig:70:

config BT_STM32WBA
    bool
    default y
    select HAS_STM32LIB
    select BT_HCI_SET_PUBLIC_ADDR
    depends on DT_HAS_ST_HCI_STM32WBA_ENABLED && ZEPHYR_HAL_STM32_MODULE_BLOBS && BT_HCI && BT_DRIVERS
    help
      ST STM32WBA HCI Bluetooth interface
Running KconfigBasicNoModules tests in /home/sylvio/espressif/zephyr/zephyr ...
warning: undefined symbol ZEPHYR_HAL_STM32_MODULE_BLOBS:
```

**Environment (please complete the following information):**
 - OS: Linux
 - 0.16.8
 -  17545c17a508fff54e0f937a76b6abc938c60c44

### PR DESCRIPTION
**Pull request by jhedberg**
There are HCI drivers which reference the module-specific Kconfig symbols which are auto-generated when the modules are present. Provide default definitions for these symbols in case the modules are missing, so that compliance checking doesn't stumble over seemingly non-existent Kconfig options.

Fixes #78056

### CODE REVIEW

**Review by erwango — COMMENTED** [review]
Fine to me but can this be documented somehow ?

**Comment by jhedberg** [comment]
> Fine to me but can this be documented somehow ?

@erwango I added a second commit for this.

**Review by jerome-pouiller — DISMISSED** [review]

File: doc/develop/modules.rst [hunk 1]
```diff
@@ -621,6 +621,14 @@ Zephyr modules
 In both Kconfig and CMake, the variable ``ZEPHYR_<MODULE_NAME>_MODULE_DIR``
 contains the absolute path to the module.
 
+Additionally, ``ZEPHYR_<MODULE_NAME>_MODULE`` and ``ZEPHYR_<MODULE_NAME>_MODULE_BLOBS``
+(in case the module declares blobs) symbols are automatically generated for available
+modules. These can be used e.g. do declare dependencies from other Kconfig symbols
```
Comment (kartben) [inline doc/develop/modules.rst:626]:
```suggestion
modules. These can be used e.g. to declare dependencies from other Kconfig symbols
```

**Review by sylvioalves — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm.

Now that we always have to define the symbol `ZEPHYR_<name>_MODULE` and `ZEPHYR_<name>_MODULE_BLOBS` then I wonder if we should revert this commit ?
https://github.com/zephyrproject-rtos/zephyr/pull/77476/commits/90f508defe0ca8ec4e4fb9b0db54038db8f43dfa

**Comment by jhedberg** [comment]
> Now that we always have to define the symbol `ZEPHYR_<name>_MODULE` and `ZEPHYR_<name>_MODULE_BLOBS` then I wonder if we should revert this commit ?
> [90f508d](https://github.com/zephyrproject-rtos/zephyr/commit/90f508defe0ca8ec4e4fb9b0db54038db8f43dfa)

Well, I used the word "recommended" in the documentation that I added, so not sure. Also, it's a bit unclear to me where exactly that commit helped, considering that we had the need for this PR to make compliance pass (on downstream trees which may omit certain modules).

**Pull request merged**